### PR TITLE
Allow for new user registrations

### PR DESCRIPTION
**Pull request by joshk**
This is built upon #1365 

Once we have this merged in we can start allowing people to sign up for NervesCloud.

### CODE REVIEW

File: lib/nerves_hub/application.ex [hunk 1]
```diff
@@ -12,6 +12,10 @@ defmodule NervesHub.Application do
         raise "fwup could not be found in the $PATH. This is a requirement of NervesHubWeb and cannot start otherwise"
     end
 
+    :logger.add_handler(:my_sentry_handler, Sentry.LoggerHandler, %{
+      config: %{metadata: [:file, :line]}
+    })
```
Comment (jjcarstens) [inline lib/nerves_hub/application.ex]: Another accidental?

File: lib/nerves_hub_web.ex [hunk 2]
```diff
@@ -47,6 +47,10 @@ defmodule NervesHubWeb do
         |> Enum.filter(fn x -> !is_nil(params[to_string(x)]) end)
         |> Enum.into(%{}, fn x -> {x, params[to_string(x)]} end)
       end
+
+      def atomize_params(params) do
+        for {key, val} <- params, into: %{}, do: {String.to_atom(key), val}
+      end
```
Comment (jjcarstens) [inline lib/nerves_hub_web.ex:53]: Same here?

File: priv/repo/migrations/20240701004550_change_username_to_name.exs
Comment (jjcarstens) [inline priv/repo/migrations/20240701004550_change_username_to_name.exs:1]: I think this is left over from the previous pr

**Review by jjcarstens — COMMENTED** [review]
I don't think the register route is in these changes?

**Comment by joshk** [comment]
> I don't think the register route is in these changes?

they were never removed: https://github.com/nerves-hub/nerves_hub_web/blob/allow-for-new-user-registrations/lib/nerves_hub_web/router.ex#L193-L194

File: lib/nerves_hub_web/controllers/account_controller.ex [hunk 3]
```diff
@@ -3,9 +3,31 @@ defmodule NervesHubWeb.AccountController do
 
   alias Ecto.Changeset
   alias NervesHub.Accounts
-  alias NervesHub.Accounts.SwooshEmail
+  alias NervesHub.Accounts.{User, SwooshEmail}
```
Comment (jjcarstens) [inline lib/nerves_hub_web/controllers/account_controller.ex:6]:
```suggestion
  alias NervesHub.Accounts.SwooshEmail
  alias NervesHub.Accounts.User
```
One thing we've been doing recently is actually removing alias expansions when we run into them. Its not a huge deal, but makes grepping/finding a little bit easier when future us has to poke around and change, so we opt for a few more lines as the tradeoff.

Not a huge deal, but food-for-thought

**Review by jjcarstens — APPROVED** [review]

File: lib/nerves_hub_web/controllers/account_controller.ex (hunk 3 above)
Comment (joshk) [reply lib/nerves_hub_web/controllers/account_controller.ex:6]: cool, I'll keep this in mind and switch them over as I come across them

**Pull request merged**